thirdeye/thirdeye-frontend/app/pods/components/rootcause-custom-baseline/component.js

```diff
@@ -26,10 +22,11 @@ export default Component.extend({
 
   selectedBaselineType: 'Unit over x units',
 
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
   baselineTypes: [UNIT_OVER_UNIT, MEAN_UNIT, MEDIAN_UNIT, MAX_UNIT, MIN_UNIT],
 
   selectedTimeUnit: 'Week',
-
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
```

Comment:
You can set a `TIME_UNITS` constant above the component declaration and reference it here so you don't have to bypass eslint.

Reply:
I have checked that nearly all components have objects and arrays assigned to the property in the ember class. Because we will refactor all ember classes to native classes after upgrading to ember octan, it is safe to assign objects to fields in the native class (see [here](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/avoid-leaking-state-in-ember-objects.md)). I make a trade-off for skipping the ember object check currently and refactor all after migrating to ember octane. it will reduce repetitive work. We can club them in a constant folder to reduce some duplicate snippets or make it a constant in the file.  Let me know your opinions on the plan.

Reply:
I am not sure we will be updating the native components any time soon. I believe we want to update them as we go considering that the codebase is pretty sizable. But yeah since it is okay for a class to be having this and since each component ends up creating its own instance, the state should not be leaked. We can keep it the way it is.